app/components/MintBTC.tsx

```diff
@@ -93,64 +151,109 @@ export function MintBTC() {
 	});
 
 	const { handleSubmit, watch, setValue } = mintNBTCForm;
-	const numberOfBTC = watch("numberOfBTC");
 
-	const youReceive = parseBTC(numberOfBTC || "0") - nBTCMintFeeInSatoshi;
+	const numberOfBTC = watch("numberOfBTC");
+	const SUIAddress = watch("suiAddress");
+	// const youReceive = parseBTC(numberOfBTC || "0") - nBTCMintFeeInSatoshi;
 
 	return (
 		<FormProvider {...mintNBTCForm}>
 			<form
-				onSubmit={handleSubmit((formData) => {
-					// TODO: handle the nbtc form data
-					console.log("Depositing with Sui Address:", formData);
+				onSubmit={handleSubmit(async ({ numberOfBTC, suiAddress }) => {
+					const isMainNetMode = isMainNetNetwork();
+					if (currentAddress) {
+						const response = await nBTCMintTxn(
+							currentAddress,
+							Number(parseBTC(numberOfBTC)),
+							remove0xPrefix(suiAddress),
+							isMainNetMode ? networks.bitcoin : networks.testnet,
```

Comment:
can we have it as a function that will just return the value here? we already have the if statement inside the funciton